### PR TITLE
doc(t3d-53): Add French translation of 'how to evolve application cicd'

### PR DESCRIPTION
**Pull request by emmanuelquinton**

### CODE REVIEW

File: content/posts/how-to-evolve-application-cicd/index.fr.md [hunk 1]
```diff
@@ -0,0 +1,337 @@
...
+
+
+## Mise en place de notre CICD
+
+
+### Notre pipeline souhaité
+
+A travers notre premier pipeline nous voulons valider :
+
+* construire l'application
```
Comment (dcremonini) [inline content/posts/how-to-evolve-application-cicd/index.fr.md:71]:
```suggestion
* construire l'application ;
```

File: content/posts/how-to-evolve-application-cicd/index.fr.md [hunk 2]
```diff
@@ -0,0 +1,337 @@
...
+* vérifier que le code est formaté
```
Comment (dcremonini) [inline content/posts/how-to-evolve-application-cicd/index.fr.md:72]:
```suggestion
* vérifier que le code est formaté ;
```

File: content/posts/how-to-evolve-application-cicd/index.fr.md [hunk 3]
```diff
@@ -0,0 +1,337 @@
...
+
+
+#### Création d'un job
+
+CircleCI définit des "jobs" qui sont des tâches exécutables ; chaque job peut être divisé en étapes
+Dans la première version de notre CI (cf. code ci-dessous) nous voulons :
+* construire le fichier jar
+* Exécuter les tests unitaires
+* Exécuter les tests d'intégration
+  Exécuter les tests d'intégration * Exécuter les tests système
```
Comment (dcremonini) [inline content/posts/how-to-evolve-application-cicd/index.fr.md:108]:
```suggestion
  * Exécuter les tests d'intégration ;
  * Exécuter les tests système.
```

File: content/posts/how-to-evolve-application-cicd/index.fr.md [hunk 4]
```diff
@@ -0,0 +1,337 @@
...
+
+![circleci-create-settings.png](circleci-create-settings.png)
+
+
+
+#### Création d'un job
+
+CircleCI définit des "jobs" qui sont des tâches exécutables ; chaque job peut être divisé en étapes
+Dans la première version de notre CI (cf. code ci-dessous) nous voulons :
+* construire le fichier jar
```
Comment (dcremonini) [inline content/posts/how-to-evolve-application-cicd/index.fr.md:105]:
```suggestion
* construire le fichier jar ;
```

File: content/posts/how-to-evolve-application-cicd/index.fr.md [hunk 5]
```diff
@@ -0,0 +1,337 @@
...
+Action : "Les actions sont des tâches individuelles que vous pouvez combiner pour créer des tâches et personnaliser votre flux de travail." cf. https://docs.github.com/fr/actions/creating-actions/about-custom-actions
+
+#### Première connexion
+
+GitHub Actions est l'outil de CI de GitHUb.
+Ce CI est accessible depuis le bouton Action dans une barre d'outils
+![github_action_menu.png](github_action_menu.png)
+
+Lors du premier accès, GithUb Actions affiche une proposition de workflow en fonction de votre projet.
+Dans notre cas, nous avons sélectionné le workflow java avec gradle.
```
Comment (dcremonini) [inline content/posts/how-to-evolve-application-cicd/index.fr.md:262]:
```suggestion
Dans notre cas, nous avons sélectionné le workflow Java avec gradle.
```

File: content/posts/how-to-evolve-application-cicd/index.fr.md [hunk 6]
```diff
@@ -0,0 +1,337 @@
...
+        cache: gradle
+    - run: ./gradlew assemble --no-daemon
+
+````
+
+
+
+Dans le code ci-dessus :
+
+* nous créons le job assemble qui est basé sur ubuntu-latest
```
Comment (dcremonini) [inline content/posts/how-to-evolve-application-cicd/index.fr.md:315]:
```suggestion
* Nous créons le job assemble qui est basé sur ubuntu-latest
```

File: content/posts/how-to-evolve-application-cicd/index.fr.md [hunk 7]
```diff
@@ -0,0 +1,337 @@
...
+<!--more-->
+
+
+## Préambule
+
+Dans le dernier article, nous avons décrit les tests ajoutés à notre projet.
+Ces tests sont utiles pour vérifier notre développement localement puisque nous utilisons le TDD. Mais ils sont aussi utiles pour vérifier s'il y a des régressions.
+Dans cet article, nous allons
+
+* définir ce qu'est le CICD
```
Comment (dcremonini) [inline content/posts/how-to-evolve-application-cicd/index.fr.md:32]:
```suggestion
* définir ce qu'est le CICD ;
```

File: content/posts/how-to-evolve-application-cicd/index.fr.md [hunk 8]
```diff
@@ -0,0 +1,337 @@
...
+## Quels outils CICD ?
+
+Il existe quelques outils pour exécuter le CICD ; nous pouvons citer :
+* Gitlab-ci
+* CircleCI
+* GitHub Action
+* Jenkins
+
+Pour les projets open source comme le nôtre, il est plus facile d'utiliser un outil de CICD intégré dans un service d'hébergement et de gestion de développement logiciel comme GitLab ou GitHub.
+C'est pourquoi nous préférons utiliser GitLab, CircleCi ou GitHub Actions.
```
Comment (dcremonini) [inline content/posts/how-to-evolve-application-cicd/index.fr.md:53]:
```suggestion
Nous préférons utiliser GitLab, CircleCi ou GitHub Actions car Jenkins nécessite un déploiement, une gestion et des ressources matérielles distincts.
```

File: content/posts/how-to-evolve-application-cicd/index.fr.md [hunk 9]
```diff
@@ -0,0 +1,337 @@
...
+car Jenkins nécessite un déploiement, une gestion et des ressources matérielles distincts.
```
Comment (dcremonini) [inline content/posts/how-to-evolve-application-cicd/index.fr.md:54]: I'd delete this line.

File: content/posts/how-to-evolve-application-cicd/index.fr.md [hunk 10]
```diff
@@ -0,0 +1,337 @@
...
+* Nous exécutons la tâche gradle assemble.
+
+
+
+
+## Résumé
+Le premier objectif de cet article était de tester trois moteurs CICD (GitLab, CircleCi et GitHub Actions), mettant en oeuvre des pipelines similaires.
+En essayant d'implémenter le CICD sur ces moteurs, nous avons rapidement rencontré quelques limitations pour les deux premiers CICD :
+
+* pousser le rapport de l'IC Gitlab vers GitHub n'est pas possible.
```
Comment (dcremonini) [inline content/posts/how-to-evolve-application-cicd/index.fr.md:326]:
```suggestion
* Pousser le rapport de la CI Gitlab vers GitHub n'est pas possible.
```

File: content/posts/how-to-evolve-application-cicd/index.fr.md [hunk 11]
```diff
@@ -0,0 +1,337 @@
...
+
+
+## Préambule
+
+Dans le dernier article, nous avons décrit les tests ajoutés à notre projet.
+Ces tests sont utiles pour vérifier notre développement localement puisque nous utilisons le TDD. Mais ils sont aussi utiles pour vérifier s'il y a des régressions.
+Dans cet article, nous allons
+
+* définir ce qu'est le CICD
+* Comment ajouter les paramètres CICD à notre projet
```
Comment (dcremonini) [inline content/posts/how-to-evolve-application-cicd/index.fr.md:33]:
```suggestion
* montrer comment ajouter les paramètres CICD à notre projet
```

File: content/posts/how-to-evolve-application-cicd/index.fr.md [hunk 12]
```diff
@@ -0,0 +1,337 @@
...
+### Notre pipeline souhaité
+
+A travers notre premier pipeline nous voulons valider :
+
+* construire l'application
+* vérifier que le code est formaté
+* vérifier la non-régression du code par l'exécution de tests.
+
+
+Nous voulons également que le pipeline soit exécuté à chaque commit sur la branche develop et les branches ci (avec le préfixe ci_).
```
Comment (dcremonini) [inline content/posts/how-to-evolve-application-cicd/index.fr.md:76]:
```suggestion
Nous voulons également que le pipeline soit exécuté à chaque commit sur la branche _develop_ et les branches CI (avec le préfixe ci_).
```

**Review by dcremonini — CHANGES_REQUESTED** [review]

File: content/posts/how-to-evolve-application-cicd/index.fr.md [hunk 13]
```diff
@@ -0,0 +1,337 @@
...
+Nous pourrions ajouter d'autres types de branches : feature(feat_), fix (fix_), documentation (doc_)
```
[the remark this answers is not in the document]
Comment (emmanuelquinton) [reply content/posts/how-to-evolve-application-cicd/index.fr.md:77]: J'ai pas l'habitude d'utiliser feat/

File: content/posts/how-to-evolve-application-cicd/index.fr.md (hunk 1 above)
Comment (emmanuelquinton) [reply content/posts/how-to-evolve-application-cicd/index.fr.md:71]: Plutôt un point

File: content/posts/how-to-evolve-application-cicd/index.fr.md (hunk 8 above)
Comment (emmanuelquinton) [reply content/posts/how-to-evolve-application-cicd/index.fr.md:53]: Done

File: content/posts/how-to-evolve-application-cicd/index.fr.md (hunk 13 above)
[the remark this answers is not in the document]
Comment (emmanuelquinton) [reply content/posts/how-to-evolve-application-cicd/index.fr.md:77]: done

File: content/posts/how-to-evolve-application-cicd/index.fr.md (hunk 1 above)
Comment (emmanuelquinton) [reply content/posts/how-to-evolve-application-cicd/index.fr.md:71]: done

File: content/posts/how-to-evolve-application-cicd/index.fr.md (hunk 2 above)
Comment (emmanuelquinton) [reply content/posts/how-to-evolve-application-cicd/index.fr.md:72]: updated with dote

File: content/posts/how-to-evolve-application-cicd/index.fr.md (hunk 3 above)
Comment (emmanuelquinton) [reply content/posts/how-to-evolve-application-cicd/index.fr.md:108]: Done with dote

File: content/posts/how-to-evolve-application-cicd/index.fr.md (hunk 4 above)
Comment (emmanuelquinton) [reply content/posts/how-to-evolve-application-cicd/index.fr.md:105]: Done

File: content/posts/how-to-evolve-application-cicd/index.fr.md (hunk 10 above)
Comment (emmanuelquinton) [reply content/posts/how-to-evolve-application-cicd/index.fr.md:326]: Done

File: content/posts/how-to-evolve-application-cicd/index.fr.md (hunk 11 above)
Comment (emmanuelquinton) [reply content/posts/how-to-evolve-application-cicd/index.fr.md:33]: Done

File: content/posts/how-to-evolve-application-cicd/index.fr.md (hunk 12 above)
Comment (emmanuelquinton) [reply content/posts/how-to-evolve-application-cicd/index.fr.md:76]: Done

**Comment by dcremonini** [comment]
I pushed some fixes.
Can you have a look?

**Review by dcremonini — APPROVED** [review]

**Pull request merged**